### PR TITLE
added new metric: contribution-distance

### PR DESCRIPTION
**Pull request by JustinGOSSES**
Added new metric: contribution-distance

Attempted to add it to the mermaid.js diagram as well but couldn't get the mermaid diagram to generate locally with or without changes, so that might need a review by @jeffabailey

### CODE REVIEW

**Comment by jeffabailey** [comment]
@JustinGOSSES I ran the script locally, and it works fine.

You don't need to update use_gqm.md though.

Here's what the graph will look like after GitHub actions generates the diagram and commits the change after the PR is merged.

Basically, adding new goals, questions, and metrics while linking the to each other is all you have to do as a contributor.

image

**Comment by JustinGOSSES** [comment]
I wanted there to be an arrow from the contribution-distance block to the question block about understanding who contributes to a project, but it looks like that arrow didn't appear?

I'll try to new generation instructions and report back.

**Comment by jeffabailey** [comment]
> I wanted there to be an arrow from the contribution-distance block to the question block about understanding who contributes to a project, but it looks like that arrow didn't appear? There is a line in the mermaid diagram code for the arrow but I'll take another look.
> 
> I'll try to new generation instructions and report back.

If you add another link to the bottom of this file and rename the metric's file to contribution-distance.md it should generate the link as expected.

https://github.com/InnerSourceCommons/managing-innersource-projects/blob/main/measuring/questions/who-uses.md?plain=1#L15

The gqm_gen script expects hyphenated filenames rather than underscore delimited.

**Comment by JustinGOSSES** [comment]
Addressed all the issues, generated diagram successfully locally, and added to the instructions for generating the diagram locally to address some problems I ran into: 
- Need to install mermaid.js globally as it doesn't get installed by the package.json
- The .mmd instead of .md file ending confusing me as I didn't expect it, so mentioned that in instructions. 
- Added some language to explicitly describe all the files that need to be possibly edited when making changes to the diagram. 
- Added a warning about not using `_` underscores in filenames and using `-` hyphens instead.

**Comment by JustinGOSSES** [comment]
image

File: scripts/gqm_gen/README.md [hunk 1]
```diff
@@ -2,11 +2,53 @@
...
+```
+
 ## Usage
 
 Builds the mermaid diagram and outputs it to stdout.
 
 ```bash
-npm install
-npm run --silent start
+npm run --silent start > gqm.mmd && mmdc --input gqm.md --output gqm.png
```
Comment (jeffabailey) [inline scripts/gqm_gen/README.md:34]: Did you need to create an .mmd file to preview with your editor?

In vscode, it's possible to preview the .md file.

image


```suggestion 
npm run --silent start > gqm.md && mmdc --input gqm.md --output gqm.png
```

File: scripts/gqm_gen/README.md [hunk 2]
```diff
@@ -2,11 +2,53 @@
...
 ```
+
+Note that the file ending `.mmd` is not a typo. It is mermaid markdown file.
+
+This should generate a file called `gqm.png` or `gqm-1.png` in the same folder as this file.
+
+You will also end up with BOTH a `gqm.md` and a `gqm.mmd` file as the later is created upon running the script.
```
Comment (jeffabailey) [inline scripts/gqm_gen/README.md:41]: No mmd file should be generated after adjusting the recommended command above.

image

File: scripts/gqm_gen/gqm-1.png
Comment (jeffabailey) [inline scripts/gqm_gen/gqm-1.png:1]: Need to delete this file, and we may want to add this to the bottom of the .gitignore file.

innersource-commons/managing-innersource-projects/scripts/gqm_gen/.gitignore

```
gqm*[pngmd]
```

**Review by jeffabailey — CHANGES_REQUESTED** [review]

**Comment by rrrutledge** [comment]
Hey @JustinGOSSES, sorry I'm behind on my _GitHub_ review queue so just looking at this.  It looks like you and @jeffabailey have had a bit of back-and-forth, so just wanted to ping you to see if you can have a look at Jeff's requested changes?  Excited to get this in!

File: scripts/gqm_gen/README.md (hunk 2 above)
Comment (JustinGOSSES) [reply scripts/gqm_gen/README.md:41]: changed as suggested

File: scripts/gqm_gen/gqm.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+```mermaid
```
[the remark this answers is not in the document]
Comment (JustinGOSSES) [reply scripts/gqm_gen/gqm.md]: done

File: scripts/gqm_gen/gqm-1.png
Comment (JustinGOSSES) [reply scripts/gqm_gen/gqm-1.png:1]: Deleted

File: scripts/gqm_gen/gqm.mmd [hunk 4]
```diff
@@ -0,0 +1,37 @@
+```mermaid
```
[the remark this answers is not in the document]
Comment (JustinGOSSES) [reply scripts/gqm_gen/gqm.mmd]: Done

**Review by jeffabailey — APPROVED** [review]

**Pull request merged**